packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -29,11 +38,20 @@ export type QueuedRequestControllerEvents =
 export type QueuedRequestControllerActions =
   QueuedRequestControllerEnqueueRequestAction;
 
+export type AllowedActions =
```

Comment:
Hmm, this doesn't seem right. The controller doesn't access these actions — it's the middleware that does. So should this simply be:

``` typescript
export type QueuedRequestControllerMessenger = RestrictedControllerMessenger<
  typeof controllerName,
  QueuedRequestControllerActions,
  QueuedRequestControllerEvents,
  never,
  never
>;
```

and then the messenger for the middleware can be:

``` typescript
export type AllowedActions =
  | NetworkControllerGetStateAction
  | NetworkControllerSetActiveNetworkAction
  | NetworkControllerSetProviderTypeAction
  | NetworkControllerGetNetworkClientByIdAction
  | NetworkControllerFindNetworkClientIdByChainIdAction
  | SelectedNetworkControllerSetNetworkClientIdForDomainAction
  | AddApprovalRequest;

export type QueuedRequestMiddlewareMessenger = ControllerMessenger<
  QueuedRequestControllerActions | AllowedActions,
  QueuedRequestControllerEvents
>;
```

?

Reply:
I like this a lot. It would definitely be a lot safer. Fixed: a946fa0b9908aa9e55711adda32c2c64ffd49157 and a6711d8aef91944c6e31c10146d6a1595a1bdaea.

Also defined separate allow lists for `ControllerMessenger` vs. `MiddlewareMessenger` in `selected-network-controller`: 9ec79e56f9f5d92ef55a2a6b7afba2b43e7f18d4.

I wonder if there are other create-middleware functions that would benefit from being similarly refactored to accept a middleware-messenger in its middleware chain?

Reply:
Potentially yeah! This seems like a useful pattern for sure.